src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -89,6 +90,27 @@ public static void Encode(Span<byte> scanline, Span<byte> previousScanline, Span
                     sum += res < 128 ? res : 256 - res;
                 }
             }
+#else
+            int x = 0;
+            for (; x < bytesPerPixel;) {
```

Comment:
This is a neat refactoring! Can you rewrite it to use the standard `for loop` syntax though to improve readability. That `++x` gets lost in the body.

Reply:
I've added some comments to explicitly state the intent here so there's no confusion. Can't really use standard loop syntax without losing a bit of efficiency. Since these low level bits get called

Reply:
`while(x < bytesPerPixel)` ?